Purchases/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -44,23 +44,25 @@ class StoreKit2TransactionListener {
         self.taskHandle = nil
     }
 
-    func handle(purchaseResult: StoreKit.Product.PurchaseResult) async -> Bool {
+    /// - Returns: whether the user cancelled
+    /// - Throws: Error if purchase was not completed successfully
+    func handle(purchaseResult: StoreKit.Product.PurchaseResult) async throws -> Bool {
```

Comment:
Before I write tests for this, @aboedo does this implementation look right?

Reply:
Well I wrote tests now 🤓

Reply:
[not for this PR] having both the bool and the throws makes me wonder whether we should return an enum instead, with a case for error, a case for userCancelled, and a case for success